typegen/src/utils.rs

```diff
@@ -140,52 +155,68 @@ fn types_equal_inner(
         }
     }
 
-    let a_ty = types.resolve(a).expect("type a should exist in registry");
-    let b_ty = types.resolve(b).expect("type b should exist in registry");
-
     // Paths differ; types won't be equal then!
     if a_ty.path.segments != b_ty.path.segments {
         return false;
     }
 
-    // Names of type params don't line up, so different then!
-    if a_ty
-        .type_params
-        .iter()
-        .zip(&b_ty.type_params)
-        .any(|(a, b)| a.name != b.name)
-    {
-        return false;
-    }
+    let mut compare_fields = |a_params: &GenericsList,
+                              b_params: &GenericsList,
+                              a: &Field<PortableForm>,
+                              b: &Field<PortableForm>|
+     -> bool {
+        fn is_assoc(param: &Field<PortableForm>) -> bool {
+            param.type_name.as_ref().is_some_and(|x| x.contains("::"))
+        }
 
-    // We'll lazily extend our type params only if the shapes match.
-    let calc_params = || {
-        let a_params = a_parent_params.extend(&a_ty.type_params);
-        let b_params = b_parent_params.extend(&b_ty.type_params);
-        (a_params, b_params)
-    };
+        let equal_name = a.name == b.name;
```

Comment:
`equal_name` doesn't look to be used anywhere so just return now rather than calculating the next stuff and then returning below?

```suggestion
        if a.name == b.name {
            return false;
        }

```